### PR TITLE
chore(demo-playwright): rewrite button test to playwright

### PR DESCRIPTION
**Pull request by 40oleg**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [x] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?
Button e2e test wrote on Cypress.

Closes # [](https://github.com/taiga-family/taiga-ui/issues/4912)

## What is the new behaviour?
Button e2e test wrote on Playwright.

### CODE REVIEW

**Comment by 40oleg** [comment]
These are screenshots by cypress:
button-dropdown-1 snap
button-dropdown-2 snap
button-dropdown-3 snap

These are screenshots by playwright:
button-dropdown-1-chromium-win32
button-dropdown-2-chromium-win32
button-dropdown-3-chromium-win32

**Comment by 40oleg** [comment]
Do I have to reduce quality of images somehow to increase performance?

**Comment by 40oleg** [comment]
@waterplea Can you take a look at this pr, please?

File: projects/demo-playwright/tests/kit/button/button.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    test.use({
+        viewport: {width: 400, height: 150},
+    });
+
+    test.beforeEach(async ({page}) => {
+        await tuiGoto(page, `components/button`);
+    });
+
+    test(`dropdown`, async ({page}) => {
+        const rotatingBlock = page.locator(`tui-doc-example[heading="Rotating"]`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/button/button.spec.ts:14]:
```suggestion
        const example = new TuiDocumentationPagePO(page).getExample(`#dropdown`);
```

File: projects/demo-playwright/tests/kit/button/button.spec.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        const button = rotatingBlock.locator(`button[data-appearance=primary]`);
+
+        await button.click();
+        await button.click();
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/button/button.spec.ts:19]: There is command `scrollIntoViewIfNeeded()`.

Hoverever, no action are required)
Command `toHaveScreenshot` should do it automatically before snapshot.

**Comment by nsbarsukov** [comment]
> Do I have to reduce quality of images somehow to increase performance?

nope, it is not required

File: projects/demo-playwright/tests/kit/button/button.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (40oleg) [reply projects/demo-playwright/tests/kit/button/button.spec.ts:19]: When I skill double click() calls, the top of the page is captured, not the needed part with button (see below)
image
(button-dropdown-1-chromium-win32.png)

And only the next screenshot after the first click call looks valid
image
(button-dropdown-2-chromium-win32.png)

[the remark this answers is not in the document]
Comment (40oleg) [reply projects/demo-playwright/tests/kit/button/button.spec.ts:19]: The command you suggested works great, if you don't mind I use it to solve trouble above mentioned.

**Comment by 40oleg** [comment]
@nsbarsukov I fixed everything. Can you take a look, please?

**Review by splincode — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Comment by nsbarsukov** [comment]
@40oleg 
Thanks ❤️

**Pull request merged**